types/ember__application/index.d.ts

```diff
@@ -113,18 +120,27 @@ export default class Application extends Engine {
     buildInstance(options?: object): ApplicationInstance;
 }
 
+// Known framework objects, so that `getOwner` can always, accurately, return
+// `Owner` when working with one of these classes, which the framework *does*
+// guarantee will always have an `Owner`. NOTE: this must be kept up to date
+// whenever we add new base classes to the framework. For example, if we
+// introduce a standalone `Service` or `Route` base class which *does not*
+// extend from `EmberObject`, it will need to be added here.
+type FrameworkObject = EmberObject | GlimmerComponent;
```

Comment:
this is just union, internal, not really usefull on its own (used in one place):
```ts
export function getOwner<O extends object = object>(
    object: O,
): O extends EmberObject | GlimmerComponent ? O : Owner | undefined;
```

Reply:
Per my overall comment, your suggestion is not how we want the type to be written when we export it from Ember, so I'm going to leave this as is. Thanks.